Lib/idlelib/idle_test/test_sidebar.py

```diff
@@ -370,5 +378,164 @@ def assert_colors_are_equal(colors):
         assert_colors_are_equal(orig_colors)
 
 
+class TestShellSidebar(unittest.TestCase):
+    root: tk.Tk = None
+    shell: PyShell = None
+
+    @classmethod
+    def setUpClass(cls):
+        requires('gui')
+
+        idlelib.pyshell.use_subprocess = True
+
+        cls.root = root = tk.Tk()
+
+        fix_scaling(root)
+        fixwordbreaks(root)
+        fix_x11_paste(root)
+
+        cls.flist = flist = PyShellFileList(root)
+        macosx.setupApp(root, flist)
+        root.update_idletasks()
+
+    @classmethod
+    def tearDownClass(cls):
+        if cls.shell is not None:
+            cls.shell.executing = False
+            cls.shell.close()
+            cls.shell = None
+        cls.flist = None
+        cls.root.update_idletasks()
+        cls.root.destroy()
+        cls.root = None
+
+    @classmethod
+    def init_shell(cls):
+        cls.shell = cls.flist.open_shell()
+        cls.root.update()
+        cls.n_preface_lines = int(cls.shell.text.index('end-1c').split('.')[0]) - 1
+
+    @classmethod
+    def reset_shell(cls):
+        cls.shell.per.bottom.delete(f'{cls.n_preface_lines+1}.0', 'end-1c')
+        cls.shell.shell_sidebar.update_sidebar()
+        cls.root.update()
+
+    def setUp(self):
+        if self.shell is None:
+            self.init_shell()
+        else:
+            self.reset_shell()
+
+    def get_sidebar_lines(self):
+        canvas = self.shell.shell_sidebar.canvas
+        sleep(0.1)
+        self.root.update()
+        texts = list(canvas.find(tk.ALL))
+        texts.sort(key=lambda text: canvas.bbox(text)[1])
+        return [canvas.itemcget(text, 'text') for text in texts]
+
+    def assertSidebarLinesEndWith(self, expected_lines):
+        self.assertEqual(
+            self.get_sidebar_lines()[-len(expected_lines):],
+            expected_lines,
+        )
+
+    def getShellLineYCoords(self):
```

Comment:
Except for assert methods, please stick with PEP8 and current IDLE convention for function names.
```suggestion
    def get_shellline_ycoords(self):
```
Etcetera

Reply:
Bah, sorry, silly mistake!